include/wx/event.h

```diff
@@ -1863,31 +1872,92 @@ class WXDLLIMPEXP_CORE wxSetCursorEvent : public wxEvent
 
 private:
     wxDECLARE_DYNAMIC_CLASS_NO_ASSIGN(wxSetCursorEvent);
+};
+
+ // Touch Base Event
+
+class WXDLLIMPEXP_CORE wxTouchEventBase : public wxEvent
+{
+public:
+    wxTouchEventBase(wxWindowID winid = 0, wxEventType type = wxEVT_NULL)
+        : wxEvent(winid, type)
+    {
+    }
+
+    wxTouchEventBase(const wxTouchEventBase& event) : wxEvent(event)
+    {
+        m_pos = event.m_pos;
+    }
+
+    const wxPoint& GetPosition() const { return m_pos; }
+    void SetPosition(const wxPoint& pos) { m_pos = pos; }
+
+protected:
+    wxPoint m_pos;
+
+    wxDECLARE_ABSTRACT_CLASS(wxTouchEventBase);
+
+};
+
+// MultiTouch Event
+
+class wxTouchPointId : public wxItemId<void*>
+{
+public:
+    wxTouchPointId() : wxItemId<void*>() { }
+    wxEXPLICIT wxTouchPointId(void* pItem) : wxItemId<void*>(pItem) { }
+};
+
+class WXDLLIMPEXP_CORE wxMultiTouchEvent : public wxTouchEventBase
+{
+public:
+    wxMultiTouchEvent(wxWindowID winid = 0, wxEventType type = wxEVT_NULL)
+        : wxTouchEventBase(winid, type)
+    {
+        m_isPrimary = false;
+    }
+
+    wxMultiTouchEvent(const wxMultiTouchEvent& event) : wxTouchEventBase(event)
+    {
+        m_isPrimary = event.m_isPrimary;
+        m_sequence = event.m_sequence;
+    }
+
+    bool IsPrimary() const { return m_isPrimary; }
+    void SetPrimary(bool primary) { m_isPrimary = primary; }
+    const wxTouchPointId& GetSequence() const { return m_sequence; }
```

Comment:
This is a bit confusing, why is "sequence" an "ID"? Shouldn't a sequence consist of several IDs? If not, I suggest renaming `wxTouchPointId` to `wxTouchSequenceId' although, to be honest, I'm not sure if it's appropriate to call it an ID at all, considering that it doesn't seem to be used to identify anything in the same sense as other `wxItemID`-derived classes do.

Reply:
The ID tracks the individual fingers: Each finger can be compared to a individual cursor - in the event you need to know, which one has moved. As various plattforms have got different ways for implementing the ID, it is an opaque type just supporting copying and checking equalitity.

Reply:
OK, thanks, but why is the accessor called `GetSequence()`? Also, maybe `wxTouchPointId` should be `wxTouchCursorId` then?

Reply:
Sequence has been take from GTK. Windows calls the field dwID. OS X call it identity.
https://msdn.microsoft.com/de-de/library/windows/desktop/dd317334(v=vs.85).aspx
https://developer.apple.com/documentation/appkit/nstouch/1535399-identity

I would not call it cursor, as this is the blinking onscreen object.

Reply:
Again, my main problem is inconsistency between "point" and "sequence" which seem to be totally different to me, but are actually the same thing. I admit I don't like the use of "sequence" in this context very much neither, but I think using `wxTouchSequenceId` for the type and `GetSequenceId()` for the method would be reasonably clear.

Alternatively, we could use `wxTouchEventIdentity` and `GetEventIdentity()` if you wish, but we really shouldn't abbreviate "Identity" to "ID" as an ID already means something else in wx events context.

"Cursor" is indeed bad, I only used it because you mentioned it in the first reply, but I don't really like it neither.

To summarize: let's use "SequenceId" if you don't mind or please feel free to choose anything else but please use the same or similar names for the type and the corresponding accessor and let's try not to overload "event id" with yet another meaning.